src/_services/authentication.service.ts

```diff
@@ -1,39 +1,67 @@
-import {Injectable} from '@angular/core';
-import 'rxjs/add/operator/map';
+import {HttpClient, HttpErrorResponse} from '@angular/common/http';
+import {Inject, Injectable} from '@angular/core';
+import {Observable, of} from 'rxjs';
+import {catchError, map} from 'rxjs/operators';
 import {TokenService} from './token.service';
-import {HttpClient} from '@angular/common/http';
 import {environment} from '../environments/environment';
-import {User} from '../_models';
+import {User} from '../_models/User';
+import {AlertService} from './alert.service';
+import {ActivatedRouteSnapshot} from '@angular/router';
+import {Union, UnionMap, UNIONS_TOKEN} from '../_constants/unions';
+import {DOCUMENT} from '@angular/common';
 
-@Injectable()
+@Injectable({providedIn: 'root'})
 export class AuthenticationService {
   constructor(private http: HttpClient,
-              private tokenService: TokenService) {
+              private alertService: AlertService,
+              private tokenService: TokenService,
+              @Inject(UNIONS_TOKEN) private unions: UnionMap,
+              @Inject(DOCUMENT) private document: Document,
+  ) {
   }
 
-  login(user: User) {
-    return this.http.post(environment.backendUrl + 'token/new', user, {responseType: 'text'});
+  login(user: User): Observable<string | null> {
```

Comment:
The idea here is in returning full object with additional information, that's why token returned from backend as a JSON.

Reply:
interface Login { 
  token: string;
}

```suggestion
  login(user: User): Observable<Login  | null> {
```

kind of?

Reply:
Login can be tricky. For now, LoginResponce is the best option comes to the head.